src/App.jsx

```diff
@@ -0,0 +1,107 @@
+import './App.css';
+import {Link} from 'react-router-dom';
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+
+import Header from './component/Header';
+import BestProducts from './component/BestProducts';
+import ProductList from './component/ProductList';
+import Pagination from './component/Pagination';
+import SortMenu from './component/SortMenu';
+
+function App() {
+  const [bestProducts, setBestProducts] = useState([]); 
+  const [isOpen, setOpen] = useState(false);
+  const [sortbtntext, setSortbtntext] = useState('최신순');
+  const [sortedProducts, setSortedProducts] = useState([]);
+  const pagebuttons = [1, 2, 3, 4, 5];
+  const [activePage, setActivePage] = useState(1);
+  const [sortOrder, setSortOrder] = useState('recent'); 
+
+ 
+  useEffect(() => {
+    const fetchBestProducts = async () => {
+      try {
+        const response = await axios.get(`https://panda-market-api.vercel.app/products?page=1&orderBy=favorite`);
+        const bestProducts = response.data.list.slice(0, 4); 
+        setBestProducts(bestProducts);
+      } catch (error) {
+        console.log('Error fetching best products:', error);
+      }
+    };
+    fetchBestProducts();
+  }, []); 
+
+ 
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(`https://panda-market-api.vercel.app/products?page=${activePage}`);
+        const sortedProducts = response.data.list.sort((a, b) => {
+          return sortOrder === 'favorite'
+            ? b.favoriteCount - a.favoriteCount
+            : new Date(b.updatedAt) - new Date(a.updatedAt);
+        });
+        setSortedProducts(sortedProducts);
```

Comment:
P2:
이부분도 위와 동일하게 orderBy param을 통해서 정렬된 채로 데이터를 받으실 수 있습니다.

```suggestion
        const response = await axios.get(`https://panda-market-api.vercel.app/products?page=${activePage}&orderBy=${sortOrder}`);
        setSortedProducts(response.data.list);
```